server/src/test/java/io/crate/expression/operator/EqOperatorTest.java

```diff
@@ -96,4 +103,33 @@ public void testEvaluateEqOperator() {
         assertNormalize("1.2 = null", isLiteral(null));
         assertNormalize("'foo' = null", isLiteral(null));
     }
+
+    @Test
+    public void test_array_equals_empty_array_on_all_types() throws Exception {
+        List<DataType<?>> allTypes = new ArrayList<>();
+        DataTypeTesting.ALL_STORED_TYPES_EXCEPT_ARRAYS.forEach(type -> {
+            allTypes.add(type);
+        });
+
+        for (DataType<?> type : allTypes) {
```

Comment:
Oops, forgot to change this - used to build array type on top of each type but not needed since usage of "array" in declaration string. Pushed a fixup.